### PR TITLE
API-28023 Move appealable issues data from headers to body

### PR DESCRIPTION
**Pull request by caseywilliams**
## Summary

This moves data previously sent via headers to the body. There is only one (index) endpoint in the the Appealable Issues API, but I converted it from a GET request to a POST request because it now accepts a JSON body. I'll leave comments below with a few things to consider before merging.

## Related issue(s)
- [API-28023](https://jira.devops.va.gov/browse/API-28023)

## Testing done

- Updated specs and tested manually with local server

## What areas of the site does it impact?
- Appealable Issues API v0 (not yet public)

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by caseywilliams — COMMENTED** [review]
I'm feeling a little apprehensive about this one (because of the conversion from GET to POST). One possible alternative that occurs to me is this:

What if we instead kept this as a GET endpoint and accepted only an ICN parameter? Using this ICN, we could look up the veteran's SSN and send that along to `Caseflow::Service.new.get_contestable_issues` as identification.

The only thing that worries me about this is that I don't know whether central mail expects the file number. We can't get file number from MPI, so if there is anything special about it vs. an SSN for this purpose, we need to keep collecting the file number...

File: modules/appeals_api/app/controllers/appeals_api/appealable_issues/v0/appealable_issues_controller.rb [hunk 1]
```diff
@@ -7,25 +7,94 @@ class AppealableIssuesController < AppealsApi::V2::DecisionReviews::ContestableI
...
+    skip_before_action :validate_json_schema
+
     SCHEMA_OPTIONS = { schema_version: 'v0', api_name: 'appealable_issues' }.freeze
 
     OAUTH_SCOPES = {
-      GET: %w[veteran/AppealableIssues.read representative/AppealableIssues.read system/AppealableIssues.read]
+      GET: %w[veteran/AppealableIssues.read representative/AppealableIssues.read system/AppealableIssues.read],
+      # NOTE: the POST scopes for this API are "read" scopes because this API does not actually write any data.
+      # These are used in the index action when POSTing PII to list any matching appealable issues:
+      POST: %w[veteran/AppealableIssues.read representative/AppealableIssues.read system/AppealableIssues.read]
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/appeals_api/app/controllers/appeals_api/appealable_issues/v0/appealable_issues_controller.rb:19]: I agree with your choice, at the end of the day, irregardless of the http 'verb' we are using, it's a read, nodata is written

**Comment by michelpmcdonald** [comment]
> What if we instead kept this as a GET endpoint and accepted only an ICN parameter? Using this ICN, we could look up the veteran's SSN via MPI and send that along to `Caseflow::Service.new.get_contestable_issues` as identification.
> 
> The only thing that worries me about this is that I don't know whether central mail expects the file number. We can't get file number from MPI, so if there is anything special about it vs. an SSN for this purpose, we need to keep collecting the file number...

I like this idea, add icn as a url parameter(just confirmed with drew, that at this point in time using icn as a url parameter is ok, however, drew now seems to be involved in some discussion that ICN is in fact consider PII, so this might be subject to change, check back with drew before implementing i guess??)

we can get file number from bgs:
`bgss = BGS::Services.new(external_uid: ssn, external_key: ssn)
p = bgss.people.find_by_ssn(ssn)
p.file_nbr
`

**Comment by caseywilliams** [comment]
Closing this in favor of #13521

**Pull request closed without merging**